python/GafferUI/SpreadsheetUI.py

```diff
@@ -0,0 +1,1560 @@
+	# TweakPlug can provide more meaningful names than "name" or "enabled" when their child
+	# plugs are added to spreadsheets.
+	columnName = Gaffer.Metadata.value( plug, "spreadsheet:columnName" ) or plug.getName()
+
+	columnIndex = spreadsheet["rows"].addColumn( plug, columnName )
+	valuePlug = spreadsheet["rows"]["default"]["cells"][columnIndex]["value"]
+	Gaffer.MetadataAlgo.copy( plug, valuePlug, exclude = "spreadsheet:columnName layout:* deletable" )
+
+	return columnIndex
+
+def __createSpreadsheet( plug ) :
+
+	spreadsheet = Gaffer.Spreadsheet()
+	__addColumn( spreadsheet, plug )
+	spreadsheet["rows"].addRow()
+
+	with Gaffer.UndoContext( plug.ancestor( Gaffer.ScriptNode ) ) :
+		plug.node().parent().addChild( spreadsheet )
+		plug.setInput( spreadsheet["out"][0] )
+
+	GafferUI.NodeEditor.acquire( spreadsheet )
+
+def __addToSpreadsheet( plug, spreadsheet ) :
+
+	with Gaffer.UndoContext( spreadsheet.ancestor( Gaffer.ScriptNode ) ) :
+		columnIndex = __addColumn( spreadsheet, plug )
+		plug.setInput( spreadsheet["out"][columnIndex] )
+
+def __addToSpreadsheetSubMenu( plug ) :
+
+	menuDefinition = IECore.MenuDefinition()
+
+	alreadyConnected = []
+	other = []
+	for spreadsheet in Gaffer.Spreadsheet.Range( plug.node().parent() ) :
+
+		connected = False
+		for output in spreadsheet["out"] :
+			for destination in output.outputs() :
+				if destination.node() == plug.node() :
+					connected = True
+					break
+			if connected :
+				break
+
+		if connected :
+			alreadyConnected.append( spreadsheet )
+		else :
+			other.append( spreadsheet )
+
+	if not alreadyConnected and not other :
+		menuDefinition.append(
+			"/No Spreadsheets Available",
+			{
+				"active" : False,
+			}
+		)
+		return menuDefinition
+
+	alreadyConnected.sort( key = Gaffer.GraphComponent.getName )
+	other.sort( key = Gaffer.GraphComponent.getName )
+
+	def addItem( spreadsheet ) :
+
+		menuDefinition.append(
+			"/" + spreadsheet.getName(),
+			{
+				"command" : functools.partial( __addToSpreadsheet, plug, spreadsheet )
+			}
+		)
+
+	if alreadyConnected and other :
+		menuDefinition.append( "/__ConnectedDivider__", { "divider" : True, "label" : "Connected" } )
+
+	for spreadsheet in alreadyConnected :
+		addItem( spreadsheet )
+
+	if alreadyConnected and other :
+		menuDefinition.append( "/__OtherDivider__", { "divider" : True, "label" : "Other" } )
+
+	for spreadsheet in other :
+		addItem( spreadsheet )
+
+	return menuDefinition
+
+def __prependSpreadsheetCreationMenuItems( menuDefinition, plugValueWidget ) :
+
+	plug = plugValueWidget.getPlug()
+	if not isinstance( plug, Gaffer.ValuePlug ) :
+		return
+
+	node = plug.node()
+	if node is None or node.parent() is None :
+		return
+
+	if plug.getInput() is not None or not plugValueWidget._editable() or Gaffer.MetadataAlgo.readOnly( plug ) :
+		return
+
+	ancestorPlug, ancestorLabel = None, ""
+	for ancestorType, ancestorLabel in [
+		( Gaffer.TransformPlug, "Transform" ),
+		( Gaffer.Transform2DPlug, "Transform" ),
```

Comment:
If you do add the transform plug, you currently see nothing in the cell in the spreadsheet: Maybe a 'click to edit...' placeholder in grey or similar?
![image](https://user-images.githubusercontent.com/896779/70725823-fb2f4b00-1cf4-11ea-9216-ef9030428488.png)

Reply:
079f19a. On a mac today, so if you could check on Linux that would be grand.